### PR TITLE
Add apptesting to support writing unittest

### PR DESCRIPTION
**Pull request by dao-notional**
#110 
Resolve this issue

### CODE REVIEW

File: app/apptesting/test_suite.go [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+			ConsensusPubkey:   pkAny,
+			Jailed:            false,
+			Status:            stakingtypes.Bonded,
+			Tokens:            bondAmt,
+			DelegatorShares:   sdk.OneDec(),
+			Description:       stakingtypes.Description{},
+			UnbondingHeight:   int64(0),
+			UnbondingTime:     time.Unix(0, 0).UTC(),
+			Commission:        stakingtypes.NewCommission(sdk.ZeroDec(), sdk.ZeroDec(), sdk.ZeroDec()),
+			MinSelfDelegation: sdk.ZeroInt(),
```
Comment (catconcat) [inline app/apptesting/test_suite.go]: @dao-notional I think the MinSelfDelegation should be set here, the value shouldn't be zero.

Also, why the x/bank/keeper tests still pass when the MinSelfDelegaton is being set to zero. Isn't it supposed to be caught by the newly added ante?
